Xamarin.PropertyEditing/StringConversionExtensions.cs

```diff
@@ -0,0 +1,199 @@
+		{
+			return (T)Enum.Parse (typeof (T), value, true);
+		}
+
+		public static string ToEditorString (this int value)
+		{
+			return value.ToString (invariantCulture.NumberFormat);
+		}
+
+		public static string ToEditorString (this double value)
+		{
+			return value == 0d ? "0.0" : value.ToString (invariantCulture.NumberFormat);
+		}
+
+		public static string ToEditorString (this float value)
+		{
+			return value == 0f ? "0.0" : value.ToString (invariantCulture.NumberFormat);
+		}
+
+		public static int ToEditorDigitsCount (this double value)
+		{
+			var baseDigits = ((int)Math.Truncate (value)).ToString ().Length;
+			var fullDigits = value.ToString (invariantCulture.NumberFormat).Length;
+			return Math.Max (0, fullDigits - baseDigits - 1);
+		}
+
+		public static string ToEditorString (this PointF value)
+		{
+			return string.Format ("{0}, {1}",
+				value.X.ToEditorString (),
+				value.Y.ToEditorString ());
+		}
+
+		public static string ToDesignerString (this RectangleF value)
+		{
+			return string.Format ("{0}, {1}, {2}, {3}",
+				value.X.ToEditorString (),
+				value.Y.ToEditorString (),
+				value.Width.ToEditorString (),
+				value.Height.ToEditorString ());
+		}
+
+		public static string ToEditorString (this SizeF value)
+		{
+			return string.Format ("{0}, {1}",
+				value.Width.ToEditorString (),
+				value.Height.ToEditorString ());
+		}
+
+		/// <summary>
+		/// Transform a camel or Pascal cased string into one with spaces
+		/// </summary>
+		public static string ToHumanString (this object obj)
+		{
+			var str = obj.ToString ();
+			var sb = new StringBuilder (str.Length + 10);
+			sb.Append (char.ToUpper (str[0]));
+			for (int i = 1; i < str.Length; i++) {
+				var chr = str[i];
+				if (char.IsUpper (chr) && i < str.Length - 1 && (!char.IsUpper (str, i - 1) || !char.IsUpper (str, i + 1)))
+					sb.Append (' ');
+				sb.Append (chr);
+			}
+			return sb.ToString ().Trim ();
+		}
+
+		/// <summary>
+		/// Transform a Pascal case C# value to a camel case value used in storyboards
+		/// </summary>
+		public static string ToCamelCase (this string value)
+		{
+			if (string.IsNullOrEmpty (value))
+				return value;
+			var first = value[0];
+			return char.ToLowerInvariant (first) + value.Substring (1);
+		}
+
+		/// <summary>
+		/// Transform a camel case value used in storyboards to a Pascal case C# value
+		/// </summary>
+		public static string ToPascalCase (this string value)
+		{
+			if (string.IsNullOrEmpty (value))
+				return value;
+			var first = value[0];
+			return char.ToUpperInvariant (first) + value.Substring (1);
+		}
+
+		/// <summary>
+		/// Handle versions like '3035' and also '3035.24'. By default System.Version blows up if there's no '.'
+		/// </summary>
+		public static Version ToVersion (this string value)
+		{
+			if (string.IsNullOrEmpty (value))
+				return new Version ();
+			if (value.Contains ("."))
+				return Version.Parse (value);
+			return new Version (int.Parse (value), 0);
+		}
```

Comment:
Do we need all of these extension methods ported over?

Reply:
How many of these methods can we remove? Let's remove everything we don't use right now!